Flaky test `OpensearchExporterIT.shouldAddIndexLifecycleSettingsToExistingIndicesOnRerunWhenRetentionIsEnabled`

**Summary**

- How often does the test fail? [once](https://github.com/camunda/zeebe/actions/runs/8541686929/attempts/1#summary-23401588329)
- Does it block your work? no
- Do we suspect that it is a real failure? no, it's most probably due to the async nature of opensearch update policy functions

**Failures**

<details><summary>Example assertion failure</summary>
<pre>
shouldAddIndexLifecycleSettingsToExistingIndicesOnRerunWhenRetentionIsEnabled (Flaky Test): io.camunda.zeebe.exporter.opensearch.OpensearchExporterIT$IndexSettingsTest
org.awaitility.core.ConditionTimeoutException: 
Assertion condition defined as a Lambda expression in io.camunda.zeebe.exporter.opensearch.OpensearchExporterIT$IndexSettingsTest [ISM policy should not be configured] 
Expecting an empty Optional but was containing value: IndexISMPolicyDto[policyId=zeebe-record-retention-policy] within 10 seconds.
	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:167)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:119)
	at org.awaitility.core.AssertionCondition.await(AssertionCondition.java:31)
	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:1006)
	at org.awaitility.core.ConditionFactory.untilAsserted(ConditionFactory.java:790)
	at io.camunda.zeebe.exporter.opensearch.OpensearchExporterIT$IndexSettingsTest.shouldAddIndexLifecycleSettingsToExistingIndicesOnRerunWhenRetentionIsEnabled(OpensearchExporterIT.java:299)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:387)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1312)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1843)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1808)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:188)
Caused by: java.lang.AssertionError: [ISM policy should not be configured] 
Expecting an empty Optional but was containing value: IndexISMPolicyDto[policyId=zeebe-record-retention-policy]
	at io.camunda.zeebe.exporter.opensearch.OpensearchExporterIT$IndexSettingsTest.assertHasNoISMPolicy(OpensearchExporterIT.java:428)
	at io.camunda.zeebe.exporter.opensearch.OpensearchExporterIT$IndexSettingsTest.lambda$shouldAddIndexLifecycleSettingsToExistingIndicesOnRerunWhenRetentionIsEnabled$0(OpensearchExporterIT.java:302)
	at org.awaitility.core.AssertionCondition.lambda$new$0(AssertionCondition.java:53)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:248)
	at org.awaitility.core.ConditionAwaiter$ConditionPoller.call(ConditionAwaiter.java:235)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
</pre>
</details>

**Hypotheses**
it's most probably due to the async nature of opensearch update policy functions

**Logs**

https://productionresultssa7.blob.core.windows.net/actions-results/47e02c5d-1cf1-4033-adbb-3610e895ac89/workflow-job-run-65e11684-88a0-5014-4a3a-b5bb082dc98b/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-04-03T16%3A57%3A18Z&sig=QnN0pMP9k0QxAZyYQQZm2GnO7nLdjFvNE3nbQFVe%2BSw%3D&sp=r&spr=https&sr=b&st=2024-04-03T16%3A47%3A13Z&sv=2021-12-02
<details><summary>Logs</summary>
<pre>
LOGS
</pre>
</details>

--- a/zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchExporterIT.java
+++ b/zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/OpensearchExporterIT.java
@@ -280,7 +280,7 @@ final class OpensearchExporterIT {
 
   /**
    * policy change is an asynchronous background process in opensearch, that's why we use awaits
-   * before asserts to reduce flakey results
+   * before asserts to reduce flaky results
    */
   @Nested
   final class IndexSettingsTest {
@@ -296,6 +296,7 @@ final class OpensearchExporterIT {
       // then
       final var index1 = indexRouter.indexFor(record1);
       await()
+          .atMost(Duration.ofSeconds(30))
           .untilAsserted(
               () -> {
                 final var index1Policy = testClient.explainIndex(index1);
@@ -313,12 +314,14 @@ final class OpensearchExporterIT {
       // then
       final var index2 = indexRouter.indexFor(record2);
       await()
+          .atMost(Duration.ofSeconds(30))
           .untilAsserted(
               () -> {
                 final var index2Policy = testClient.explainIndex(index2);
                 assertHasISMPolicy(index2Policy);
               });
       await()
+          .atMost(Duration.ofSeconds(30))
           .untilAsserted(
               () -> {
                 final var index1PolicyNew = testClient.explainIndex(index1);
@@ -338,6 +341,7 @@ final class OpensearchExporterIT {
       // then
       final var index1 = indexRouter.indexFor(record1);
       await()
+          .atMost(Duration.ofSeconds(30))
           .untilAsserted(
               () -> {
                 final var indexPolicy1 = testClient.explainIndex(index1);
@@ -363,6 +367,7 @@ final class OpensearchExporterIT {
                 assertHasNoISMPolicy(response2);
               });
       await()
+          .atMost(Duration.ofSeconds(30))
           .untilAsserted(
               () -> {
                 final var index1PolicyNew = testClient.explainIndex(index1);
@@ -394,6 +399,7 @@ final class OpensearchExporterIT {
       // then
       final var index2 = indexRouter.indexFor(record2);
       await()
+          .atMost(Duration.ofSeconds(30))
           .untilAsserted(
               () -> {
                 final var index2Policy = testClient.explainIndex(index2);

--- a/zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/TestClient.java
+++ b/zeebe/exporters/opensearch-exporter/src/test/java/io/camunda/zeebe/exporter/opensearch/TestClient.java
@@ -190,7 +190,7 @@ final class TestClient implements CloseableSilently {
 
   void deleteIndices() {
     try {
-      final var request = new Request("DELETE", config.index.prefix + "*");
+      final var request = new Request("DELETE", config.index.prefix + "*?expand_wildcards=all");
       restClient.performRequest(request);
     } catch (final IOException e) {
       throw new UncheckedIOException(e);

